jetty-start/src/main/java/org/eclipse/jetty/start/Modules.java

```diff
@@ -215,7 +216,22 @@ private Module registerModule(Path file)
             _names.put(module.getName(), module);
             module.getProvides().forEach(n ->
             {
-                _provided.computeIfAbsent(n, k -> new HashSet<Module>()).add(module);
+                // Syntax can be :
+                // "<name>" - for a simple provider reference
+                // "<name>|default" - for a provider that is also the default implementation
```

Comment:
While I'm not opposed to a mechanism to tag a module as the default for the tag that it provides, I think it is a complication we don't need for this PR.

It would be sufficient to have a module called `logging` that just depends on `logging-jetty` and that will transitively be picked as the default unless another logging module is explicitly enabled.

Reply:
In fact I definitely would like to see this default mechanism removed for now.   I just did an experiment of removing the default and creating a `logging.mod` file instead and it did not work.  I think the logic to used the obvious named module may have been removed.... checking....

Reply:
Actually cancel this.... the existing named mechanism is still working, it is just really clumsy to use in this situation, as it needs to have a `logging-impl` target rather than just logging.  
Thus I think the new default mechanism is probably neater.... even though I don't like it being added in such a big PR.
But let's leave it for now.

However, we should review other modules that depend on the name mechanism and see if they would be better off with the default mechanism.